tools/importPlanGoals.js

```diff
@@ -0,0 +1,171 @@
+                  cleanTopicGoals.push({ topicId, goalId: currentGoalId });
+                }
+              });
+            } else // it's either timeframe or status}
+            if (currentGoalName !== '') {
+              currentGoal[column] = el[key].trim();
+              const goalNum = goalColumn[1].slice(0, 1);
+              currentGoals[goalNum] = { ...currentGoals[goalNum], ...currentGoal };
+            }
+          } else { // we have a goal name
+            currentGoalName = el[key].trim();
+            currentGoal = { name: currentGoalName };
+            currentGoalId = undefined;
+            if (currentGoalName !== '') {
+              const goalNum = goalColumn[1];
+              currentGoals[goalNum] = { ...currentGoals[goalNum], ...currentGoal };
+            }
+          }
+        }
+      });
+
+      // after each row
+      let goalId;
+      currentGoals.forEach((goal) => {
+        cleanGoals.push(goal);
+        goalId = cleanGoals.length;
+        currentGranteeId = getItemId(cleanGrantees, currentGrantee.trim());
+        currentGrants.forEach((grant) => {
+          const plan = { granteeId: currentGranteeId, grant, goalId };
+          if (!cleanTtaPlans.some((e) => e.granteeId === currentGranteeId
+                            && e.grant === grant
+                            && e.goalId === goalId)) {
+            cleanTtaPlans.push(plan);
+          }
+        });
+      });
+    });
+
+    // The data has been prepared. Insert it into the database
+    await Role.bulkCreate(cleanRoles.map((role, index) => (
+      { id: index + 1, name: role })),
+    {
+      updateOnDuplicate: ['name', 'updatedAt'],
+    });
+    await Topic.bulkCreate(cleanTopics.map((topic, index) => (
+      { id: index + 1, name: topic })),
+    {
+      updateOnDuplicate: ['name', 'updatedAt'],
+    });
+    await RoleTopic.bulkCreate(cleanRoleTopics,
+      {
+        ignoreDuplicates: true,
+      });
+    await Goal.bulkCreate(cleanGoals.map((goal, index) => (
+      { id: index + 1, ...goal })),
+    {
+      updateOnDuplicate: ['name', 'updatedAt'],
+    });
+    await TopicGoal.bulkCreate(cleanTopicGoals, {
+      ignoreDuplicates: true,
+    });
+    await Grantee.bulkCreate(cleanGrantees.map((grantee, index) => (
+      { id: index + 1, name: grantee })),
+    {
+      updateOnDuplicate: ['name', 'updatedAt'],
+    });
+    await Ttaplan.bulkCreate(cleanTtaPlans,
+      {
+        ignoreDuplicates: true,
+      });
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+const { argv } = option('file', {
+  alias: 'f',
+  description: 'Input .csv file',
+  type: 'string',
+})
+  .help()
+  .alias('help', 'h');
+
+const defaultInputFile = './GranteeTTAPlan.csv';
+
+let file;
+if (argv.file) {
+  // console.log('The current file is: ', argv.file);
+  file = argv.file;
+}
+
+const csv = readFileSync(file || defaultInputFile);
+[, ...grantees] = parse(csv, {
+  relax_column_count: true,
+  columns: true,
+});
+
+importGoals();
```

Comment:
Can you add tests that processes a small CSV (maybe 2 rows?) and asserts the DB looks as expected after `importGoals` runs?

Reply:
I definitely want to see a test on this, and agree with doing 2ish rows.

Reply:
Tests added.